connect/mirror/src/test/java/org/apache/kafka/connect/mirror/MirrorConnectorsIntegrationTest.java

```diff
@@ -150,23 +156,67 @@ public void setup() throws IOException {
+        // flaky tests where the connector and tasks didn't start up in time for the tests to be
+        // run
         backup.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorSourceConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(backup,
+            "MirrorSourceConnector", MIN_TASKS).isPresent(),
+            "Timed out trying to verify connector MirrorSourceConnector was up on"
+                + " backup cluster!");
 
         backup.configureConnector("MirrorCheckpointConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorCheckpointConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(backup,
+            "MirrorCheckpointConnector", MIN_TASKS).isPresent(),
+            "Timed out trying to verify connector MirrorCheckpointConnector was up on"
+                + " backup cluster!");
 
         backup.configureConnector("MirrorHeartbeatConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("primary", "backup"),
             MirrorHeartbeatConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(backup,
+            "MirrorHeartbeatConnector", MIN_TASKS).isPresent(),
+            "Timed out trying to verify connector MirrorHeartbeatConnector was up on"
+                + " backup cluster!");
 
         primary.configureConnector("MirrorSourceConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorSourceConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(primary,
+            "MirrorSourceConnector", MIN_TASKS).isPresent(),
+            "Timed out trying to verify connector MirrorSourceConnector was up on"
+                + " primary cluster!");
 
         primary.configureConnector("MirrorCheckpointConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorCheckpointConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(primary,
+            "MirrorCheckpointConnector", MIN_TASKS).isPresent(),
+            "Timed out trying to verify connector MirrorCheckpointConnector was up on"
+                + " primary cluster!");
 
         primary.configureConnector("MirrorHeartbeatConnector", mm2Config.connectorBaseConfig(new SourceAndTarget("backup", "primary"),
             MirrorHeartbeatConnector.class));
+        TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(primary,
+            "MirrorHeartbeatConnector", MIN_TASKS).isPresent(),
+            "Timed out trying to verify connector MirrorHeartbeatConnector was up on"
+                + " primary cluster!");
+    }
+
+    private Optional<Boolean> assertConnectorAndTasksRunning(EmbeddedConnectCluster connectCluster,
```

Comment:
Unless I'm missing something, we are not actually using the boolean value anywhere. All we do is check if the optional is present. Would it be simpler to just return the boolean directly?

also nit: having `assert` in the name is misleading since there are no assertions. I would suggest `isConnectorAndTasksRunning` or `checkConnectorAndTasksRunning`

Reply:
I'll change the method name to `checkConnectorAndTasksRunning`. With respect to the boolean value, As @kkonstantine mentioned, the condition to be checking should be `TestUtils.waitForCondition(() -> assertConnectorAndTasksRunning(backup,
            "MirrorHeartbeatConnector", MIN_TASKS).orElse(false)`

However, to your point, I think it might just be simpler and more readable to return the actual boolean instead of an `Optional`. I'll make that change.